### PR TITLE
add openai custom base uri

### PR DESCRIPTION
**Pull request by lifecoder1988**
test code 

image
image

### CODE REVIEW

File: src/lib/providers.ts [hunk 1]
```diff
@@ -16,15 +17,20 @@ export const getAvailableProviders = async () => {
           openAIApiKey,
           modelName: 'gpt-3.5-turbo',
           temperature: 0.7,
```
Comment (Swiftyos) [inline src/lib/providers.ts:19]: Can you add the baseURL here as well please

File: src/lib/providers.ts [hunk 2]
```diff
@@ -16,15 +17,20 @@ export const getAvailableProviders = async () => {
...
           modelName: 'gpt-4',
           temperature: 0.7,
+          configuration: {
+            baseURL: openAIBaseUrl,
+          },
         }),
         embeddings: new OpenAIEmbeddings({
           openAIApiKey,
           modelName: 'text-embedding-3-large',
+
```
Comment (Swiftyos) [inline src/lib/providers.ts:33]: Can you add another endpoint param - embedding baseURL as well as an openAIEmbeddingsAPIKey here as well please. 
 
 This is so we can use another provider for the chat model whilst keeping using openai for the embedding model

**Review by Swiftyos — CHANGES_REQUESTED** [review]

File: src/lib/providers.ts (hunk 1 above)
Comment (lifecoder1988) [reply src/lib/providers.ts:19]: thanks for you response . 

do you mean add  baseURL in  async method like this 

`export const getAvailableProviders = async ({baseuri}) => {`

i had  add  baseURL in ChatOpenAI configuration

```

models['openai'] = {
        'gpt-3.5-turbo': new ChatOpenAI({
          openAIApiKey,
          modelName: 'gpt-3.5-turbo',
          temperature: 0.7,
          configuration : {
            baseURL : openAIBaseUrl,
          }
        }),
```

File: src/lib/providers.ts (hunk 2 above)
Comment (lifecoder1988) [reply src/lib/providers.ts:33]: sure. you are right . 

the another question is where to put baseurl config , user config.toml [API_KEYS] section is fine ?

**Comment by ItzCrazyKns** [comment]
Thanks for the PR. I noticed that this PR consists of a very minor change. As we're gearing up for the 1.0.0 release, which already includes support for custom OpenAI endpoints, the code quality and effort in this PR seem very minimal. For now, I'll close this PR.

**Comment by lifecoder1988** [comment]
> Thanks for the PR. I noticed that this PR consists of a very minor change. As we're gearing up for the 1.0.0 release, which already includes support for custom OpenAI endpoints, the code quality and effort in this PR seem very minimal. For now, I'll close this PR.

fine

**Pull request closed without merging**